pkg/k8shandler/visualization.go

```
@@ -40,46 +44,31 @@ var (
 // CreateOrUpdateVisualization reconciles visualization component for cluster logging
 func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateVisualization(proxyConfig *configv1.Proxy) (err error) {
 	if clusterRequest.cluster.Spec.Visualization == nil || clusterRequest.cluster.Spec.Visualization.Type == "" {
-		if err = clusterRequest.removeKibana(); err != nil {
-			return
+		if err := clusterRequest.removeKibana(); err != nil {
+			logrus.Error(err.Error())
+			return nil
 		}
 		return nil
 	}
-	if clusterRequest.cluster.Spec.Visualization.Type == logging.VisualizationTypeKibana {
-
-		if err = clusterRequest.CreateOrUpdateServiceAccount(kibanaServiceAccountName, &kibanaServiceAccountAnnotations); err != nil {
-			return
-		}
-
-		if err = clusterRequest.RemoveClusterRoleBinding("kibana-proxy-oauth-delegator"); err != nil {
-			return
-		}
-
-		if err = clusterRequest.createOrUpdateKibanaService(); err != nil {
-			return
-		}
-
-		if err = clusterRequest.createOrUpdateKibanaRoute(); err != nil {
-			return
-		}
-
-		if err = clusterRequest.createOrUpdateKibanaSecret(); err != nil {
-			return
-		}
 
-		if err = clusterRequest.createOrUpdateKibanaConsoleExternalLogLink(); err != nil {
-			return
-		}
+	//TODO: Remove this in the next release after removing old kibana code completely
+	if err = clusterRequest.removeOldKibana(); err != nil {
+		logrus.Error(err.Error())
```

we should actually return the error here -- similar comment as above